python/tests/test_tables.py

```diff
@@ -2266,3 +2268,169 @@ def test_set_columns_not_implemented(self):
+            self.assertEqual(node_map[ee.child], e.child)
+            self.assertEqual(ee.metadata, e.metadata)
+        muts = []
+        sites = []
+        for k, m in enumerate(tables.mutations):
+            if m.node in nodes:
+                muts.append(k)
+                if m.site not in sites:
+                    sites.append(m.site)
+        site_map = np.repeat(-1, tables.sites.num_rows)
+        site_map[sites] = np.arange(len(sites), dtype="int32")
+        mutation_map = np.repeat(tskit.NULL, tables.mutations.num_rows + 1)
+        mutation_map[muts] = np.arange(len(muts), dtype="int32")
+        self.assertEqual(subset.sites.num_rows, len(sites))
+        for k, s in zip(sites, subset.sites):
+            ss = tables.sites[k]
+            self.assertEqual(ss, s)
+        self.assertEqual(subset.mutations.num_rows, len(muts))
+        for k, m in zip(muts, subset.mutations):
+            mm = tables.mutations[k]
+            self.assertEqual(mutation_map[mm.parent], m.parent)
+            self.assertEqual(site_map[mm.site], m.site)
+            self.assertEqual(node_map[mm.node], m.node)
+            self.assertEqual(mm.derived_state, m.derived_state)
+            self.assertEqual(mm.metadata, m.metadata)
+        migs = [i for i, mig in enumerate(tables.migrations) if mig.node in nodes]
+        self.assertEqual(subset.migrations.num_rows, len(migs))
+        for k, mig in zip(migs, subset.migrations):
+            mmig = tables.migrations[k]
+            self.assertEqual(mmig.left, mig.left)
+            self.assertEqual(mmig.right, mig.right)
+            self.assertEqual(node_map[mmig.node], mig.node)
+            self.assertEqual(pop_map[mmig.source], mig.source)
+            self.assertEqual(pop_map[mmig.dest], mig.dest)
+            self.assertEqual(mmig.time, mig.time)
+            self.assertEqual(mmig.metadata, mig.metadata)
+        self.assertEqual(tables.provenances, subset.provenances)
+
+    def test_subset_all(self):
+        # subsetting to everything shouldn't change things
+        # except the individual ids in the node tables if
+        # there are gaps
+        ts = self.get_msprime_mig_example(Ne=100, sample_size=10, seed=36574)
+        tables = ts.tables
+        tables2 = tables.copy()
+        tables2.subset(np.arange(ts.num_nodes), record_provenance=True)
+        tables.provenances.clear()
+        tables2.provenances.clear()
+        tables.individuals.clear()
+        tables2.individuals.clear()
+        tables.nodes.clear()
+        tables2.nodes.clear()
+        self.assertEqual(tables, tables2)
+
+    def test_random_subsets(self):
+        rng = np.random.default_rng(9385)
+        ts = self.get_msprime_mig_example(Ne=100, sample_size=10, seed=12396)
+        tables = ts.tables
+        for n in [2, ts.num_nodes - 10]:
+            nodes = rng.choice(np.arange(ts.num_nodes), n, replace=False)
+            self.verify_subset(tables, nodes)
+
+    def test_wf_examples(self):
```

Comment:
How about move the WF code to `self.get_wf_example`, and then in all the tests you can do something like this:
```
def get_examples():
   yield self.get_msprime_mig_example(Ne=100, sample_size=10, seed=12396)
   yield self.get_wf_example()

def test_subset_all(self):
    for ts in self.get_examples():
```

That way we'd be running the tests on more than one example, and would separate out the example generation from the tests.

Reply:
good idea!